community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ShortestPathComplexQueryAcceptanceTest.scala

```diff
@@ -0,0 +1,92 @@
+import org.neo4j.graphdb.Node
+
+class ShortestPathComplexQueryAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTestSupport {
+
+  test("allShortestPaths with complex LHS should be planned with exhaustive fallback and include predicate") {
+    setupModel()
+    val result = executeUsingCostPlannerOnly(
+      """
+        |//same idea but iterating over two collections of nodes
+        |profile match (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
+        |with kim as kimDeal, collect(charles) as charlesT, collect(joey) as joeyS
+        |unwind charlesT AS charlesThompson
+        |unwind joeyS AS joeySantiago
+        |match pathx = allShortestPaths((charlesThompson)-[*1..5]-(joeySantiago))
+        |where none (n IN nodes(pathx) where id(n) = id(kimDeal))
+        |with nodes(pathx) as nodes
+        |return nodes
+      """.stripMargin)
+    val results = result.columnAs("nodes").toList
+    println(results)
+    println(result.executionPlanDescription())
+    val ids = results(0).asInstanceOf[Seq[_]].map(n => n.asInstanceOf[Node].getId)
+    ids should be(List(0, 4, 3, 2))
+    result should use("VarLengthExpand(Into)", "AntiConditionalApply")
+  }
+
+  test("shortestPath with complex LHS should be planned with exhaustive fallback and include predicate") {
+    System.setProperty("pickBestPlan.VERBOSE","true")
+    setupModel()
+    val result = executeUsingCostPlannerOnly(
+      """
+        |profile match (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
+        |with kim as kimDeal, collect(charles) as charlesT, collect(joey) as joeyS
+        |unwind charlesT AS charlesThompson
+        |unwind joeyS AS joeySantiago
+        |match pathx = shortestPath((charlesThompson)-[*1..5]-(joeySantiago))
+        |where none (n IN nodes(pathx) where id(n) = id(kimDeal))
+        |unwind nodes(pathx) as nodes
+        |return nodes
+      """.stripMargin)
+    val results = result.columnAs("nodes").toList
+    println(results)
+    println(result.executionPlanDescription())
+    assertThat(results.length, equalTo(4))
+    result should use("VarLengthExpand(Into)", "AntiConditionalApply")
+  }
+
+  def executeUsingCostPlannerOnly(query: String) =
+    eengine.execute(s"CYPHER planner=IDP $query", Map.empty[String, Any], graph.session()) match {
+      case e:ExecutionResultWrapperFor3_0 => RewindableExecutionResult(e)
+    }
+
+  private def setupModel(): Unit = {
+    executeUsingCostPlannerOnly(
+      """
+        |//dataset creation
+        |merge (p1:Pixie {fname:'Charles'})
+        |merge (p2:Pixie {fname:'Kim'})
+        |merge (p3:Pixie {fname:'Joey'})
+        |merge (p4:Pixie {fname:'David'})
+        |merge (p5:Pixie {fname:'Paz'})
+        |merge (p1)-[:KNOWS]->(p2)-[:KNOWS]->(p3)-[:KNOWS]->(p4)-[:KNOWS]->(p5)-[:KNOWS]->(p1)
+        |return p1,p2,p3,p4,p5
```

Comment:
I guess it doesn't matter a lot, but these could be `CREATE` instead, and `RETURN` (and the comment) may be omitted.